core/src/edu/rice/comp413fall2020grey/Common/Message.java

```diff
@@ -0,0 +1,33 @@
+package edu.rice.comp413fall2020grey.Common;
+
+import java.io.Serializable;
+import java.util.Date;
+
+/**
+ * Abstract class that defines abstract message behaviour.
+ */
+public abstract class Message implements Serializable {
+
+  private final Date timestamp;
```

Comment:
I think we need to have a name for messages that defines the type, like ConnectMessage will have a name field of Connect - removes need to use instanceOf method to check what type of message was received
If this isn't necessary, what mechanism will the receiver use to determine what type of message it received?

Reply:
I added that functionality -- right now it's string identifiers but this can easily be changed to integers